### PR TITLE
Cherry pick 06537

### LINKED ISSUE #6523: Add tests for gas calculation utility functions
### Problem

The utility methods in `PrecompilePricingUtils` that calculate gas costs - `computeViewFunctionGas` and `computeGasRequirement` do not have adequate test coverage.

### Solution

Add unit tests and end-to-end tests for the above utility functions.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
Added unit tests for gas calculation utility methods

Fixes #6523 

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by Nana-EC — APPROVED** [review]

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/traceability/TraceabilitySuite.java [hunk 1]
```diff
@@ -4225,7 +4272,7 @@ HapiSpec traceabilityE2EScenario16() {
                                                     .setGas(79000)
                                                     .setRecipientContract(
                                                             spec.registry().getContractId(PRECOMPILE_CALLER))
-                                                    .setGasUsed(5232)
+                                                    .setGasUsed(5330)
```
Comment (david-bakin-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/traceability/TraceabilitySuite.java:4275]: Do we know what caused this discrepancy?  (Or is this two of those off-by-ones together?)

**Review by david-bakin-sl — APPROVED** [review]
LGTM

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/traceability/TraceabilitySuite.java (hunk 1 above)
Comment (lukelee-sl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/traceability/TraceabilitySuite.java:4275]: Not exactly but it seems we used about 100 more gas?

**Review by kimbor — APPROVED** [review]
lgtm!

**Review by Neeharika-Sompalli — APPROVED** [review]

**Pull request merged**